conans/client/installer.py

```diff
@@ -190,10 +186,12 @@ def build_package(self, node, keep_build, recorder, remotes):
             with package_layout.conanfile_write_lock(self._output):
                 set_dirty(build_folder)
                 self._prepare_sources(conanfile, pref, package_layout, conanfile_path, source_folder,
-                                      build_folder, package_folder, remotes)
+                                      build_folder, remotes)
 
         # BUILD & PACKAGE
         with package_layout.conanfile_read_lock(self._output):
+            remove_folder_raising(package_folder)
+            mkdir(package_folder)
```

Comment:
Is it needed? Sometimes we list the directories to know which packages are available (`PackageCacheLayout::conan_packages()`), maybe it is better to postpone the creation of any directory to the moment something is going to be copied there.

Reply:
removed